src/module-elasticsuite-indices/Helper/Index.php

```diff
@@ -0,0 +1,306 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Smile ElasticSuite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteIndices
+ * @author    Dmytro ANDROSHCHUK <dmand@smile.fr>
+ * @copyright 2020 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+namespace Smile\ElasticsuiteIndices\Helper;
+
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\DataObject;
+use Smile\ElasticsuiteCore\Client\Client;
+use Smile\ElasticsuiteCore\Helper\IndexSettings;
+use Smile\ElasticsuiteIndices\Block\Widget\Grid\Column\Renderer\IndexStatus;
+use Smile\ElasticsuiteIndices\Model\ResourceModel\StoreIndices\CollectionFactory as StoreIndicesCollectionFactory;
+use Smile\ElasticsuiteIndices\Model\ResourceModel\WorkingIndexer\CollectionFactory as WorkingIndexerCollectionFactory;
+use Zend_Date;
+use Zend_Date_Exception;
+
+/**
+ * Smile Index helper
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteIndices
+ * @author   Dmytro ANDROSHCHUK <dmand@smile.fr>
+ */
+class Index extends AbstractHelper
+{
+    /**
+     * How many days after an index is a ghost.
+     */
+    private const NUMBER_DAYS_AFTER_INDEX_IS_GHOST = 2;
+
+    /**
+     * Store number of seconds in a day.
+     */
+    private const SECONDS_IN_DAY = 86400;
+
+    /**
+     * Store number of seconds in a year.
+     */
+    private const SECONDS_IN_YEAR = 31536000;
+
+    /**
+     * @var array
+     */
+    private $elasticSuiteIndices = [
```

Comment:
This cannot be hardcoded and should be fetched from the elasticsuite_indices.xml files.

Eg if you add this module for CMS Pages, you will have a new index "cms_page" : https://github.com/Smile-SA/magento2-module-elasticsuite-cms-search

You can fetch the indices like this : https://github.com/Smile-SA/elasticsuite/blob/2.8.x/src/module-elasticsuite-core/Index/IndexSettings.php#L166

The names will then be the keys of the array.

**=> The same logic is written elsewhere, it should be refactored.**